src/core/server/integration_tests/saved_objects/migrations/type_registrations.test.ts

```diff
@@ -59,7 +59,7 @@ const previouslyRegisteredTypes = [
   'fleet-enrollment-api-keys',
   'fleet-preconfiguration-deletion-record',
   'graph-workspace',
-  'guided-setup-state',
+  'guided-onboarding-guide-state',
```

Comment:
Isn't the original `guided-setup-state` type registration shipping with 8.5? If so, we shouldn't be changing this value. If we want to rename the type, we need to treat it as if we are removing one type and adding another. That would require adding a new line to this test for `guided-onboarding-guide-state`, and adding `guided-setup-state` to the list of unused types in core (while still keeping it in the list here): https://github.com/elastic/kibana/blob/e0ada17a4f79d97c3605835ea563fb424e4c5030/packages/core/saved-objects/core-saved-objects-migration-server-internal/src/core/unused_types.ts#L11-L14

This way, `guided-setup-state` SOs won't be migrated when someone does an 8.5 -> 8.6 upgrade.

Reply:
Also, sort of goes without saying, but I assume we are okay not having this state data preserved on upgrade from 8.5 -> 8.6? Because the mappings were also changed, which would typically require a migration, unless you're okay just losing any objects created in 8.5, in which case you should be fine with the steps I outlined above.

Reply:
Thanks @lukeelmers! I may have prematurely moved this PR out of draft 😅 

The work shipped in 8.5 is behind a config flag and is hidden to the end user. Because of this, and the fact that the original data did not take into account the full support needed, it does not really make sense for it to be migrated.

With that said, is this the best approach? To rename the type and add the old type - `guided-setup-state` - to the list of removed types? Thanks!

Reply:
Even if the feature was behind a config flag in 8.5, if you don't add the old `guided-setup-state` to the list of unused types, then any deployment which had this flag enabled at some point will have migration failures when attempting to upgrade to 8.6, because it would be considered an "unknown type" that core doesn't recognize.

This can technically be worked around with the [`migrations.discardUnknownObjects`](https://www.elastic.co/guide/en/kibana/current/resolve-migrations-failures.html#unknown-saved-object-types) setting, but the user has to be aware of it.

I'm guessing the only risk here is to dev/QA environments, but as it's an easy thing to update core's unused types list, I'd probably recommend it here just to ensure nobody runs into unexpected behavior. That way the `guided-setup-state` objects will simply be left behind on upgrade from 8.5 to 8.6+, and no manual intervention is required.

Reply:
👍 thanks for the help here @lukeelmers! I've updated `unused_types` file. Would you mind taking another look when you have a minute?